mllib/src/main/scala/org/apache/spark/ml/tree/Split.scala

```diff
@@ -85,7 +85,7 @@ final class CategoricalSplit private[ml] (
   /**
    * If true, then "categories" is the set of categories for splitting to the left, and vice versa.
    */
-  private val isLeft: Boolean = _leftCategories.length <= numCategories / 2
+  private[ml] val isLeft: Boolean = _leftCategories.length <= numCategories / 2
 
   /** Set of categories determining the splitting rule, along with [[isLeft]]. */
   private val categories: Set[Double] = {
```

Comment:
We use `BitSet` in our system. I know `BitSet` takes fast constant, while `HashSet` takes effectively longer constant time depending on the length of elements. Also, `HashSet` will box the element, and `BitSet` stores as primitive elements. But I don't benchmark myself. Agree, this can be a separate PR.